### ISSUE TITLE
Forms: form validation does not work when JavaScript is disabled

### ISSUE THREAD

**Issue by talldan** [Automattic/jetpack#41942]
### Impacted plugin

Jetpack

### Quick summary

The Form block (and possibly shortcode) usually performs JavaScript form validation. See the [accessible-form.js](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/forms/src/contact-form/js/accessible-form.js) file.

There's also a lot of PHP code for form validation, which is (I assume) for when JavaScript is disabled. See the `Contact_Form_Field->validate` method:
https://github.com/Automattic/jetpack/blob/7a3559d2f7bd0cd8915902a4d0f315167793e5cd/projects/packages/forms/src/contact-form/class-contact-form-field.php#L204-L266

This PHP validation doesn't seem to work, at least on a block theme. Submitting a form that has validation errors sends the user to a 404 page.

If this needs to be supported, then it looks like something to be fixed. Alternatively, the code could be removed if form validation with JS disabled isn't a feature that needs to be kept.

### Steps to reproduce

1. Create a form with at least one required field in a post and view the post
2. In the browser dev tools, disable JS then reload the page to make sure that take effect
3. Submit the form

Expected: The form shows validation errors
Actual: A 404 page is shown. This seems to be because `?page=1` is appended to the URL.
<img width="606" alt="Image" src="https://github.com/user-attachments/assets/5e9afcf4-6c60-4161-b48e-64f7001ef8c5" />

### Site owner impact

Fewer than 20% of the total website/platform users

### Severity

Minor

### What other impact(s) does this issue have?

No revenue impact

### Platform (Simple and/or Atomic)

Self-hosted

**Comment by talldan** [comment]
The first issue is very easy to fix. There's a second issue, it now takes a user two submissions for the validation messages to show, and the form will be reset the first time. I think this is due to the form's internal id that it uses for state tracking changing after the submission. Essentially there's then a second instance of the form in state even though it's the same form.

Edit: It's actually that the form rendering happens twice, so it ends up being like there are two forms with mismatched ids, the `process_form_submission` method seems to be the culprit, here's the stack trace:
<img width="594" alt="Image" src="https://github.com/user-attachments/assets/ee908039-cf99-4549-b122-4fd968ddd838" />

**Comment by talldan** [comment]
I see this bug seems to have been introduced fairly recently via #40998. Basically the incrementing id means forms aren't idempotent, if the same form is run through the rendering code twice it'll receive a different id and cause a mismatch..

@simison just making you aware of this issue. It seems very low priority to me and I can push a small fix to solve the 404, but I don't plan to look much deeper. I just need to be able to test that the block based field has parity with the shortcode for #41840, and this bug was blocking me.

That said, if the form ids are brittle, they might cause other problems. I personally think a more solid solution is to use a uuid style id that's stored with the form markup, but I might not be aware of some of the constraints.

**Comment by talldan** [comment]
Another thing related to this that I've discovered on my journey through the form block. If using the form block in a post, the form submission expects the form id to be the post id. This code here is passing the `$id` into functions like `is_post_publicly_viewable`, `current_user_can` and `post_password_required`:
https://github.com/Automattic/jetpack/blob/49183ea8f03c179546b47bd1424933a9e2326234/projects/packages/forms/src/contact-form/class-contact-form-plugin.php#L732-L740

That `$id` variable is retrieved here:
https://github.com/Automattic/jetpack/blob/49183ea8f03c179546b47bd1424933a9e2326234/projects/packages/forms/src/contact-form/class-contact-form-plugin.php#L621

This isn't going to work so well with the way the id is now a lot more complex when initially set:
https://github.com/Automattic/jetpack/blob/49183ea8f03c179546b47bd1424933a9e2326234/projects/packages/forms/src/contact-form/class-contact-form.php#L126

It's really surprising that the id is being suddenly converted to a post id this way.

**Comment by talldan** [comment]
Discovered another check that's very similar to this here as well:
https://github.com/Automattic/jetpack/blob/6deb2860206736cfdf2279be79b132693277cbe0/projects/packages/forms/src/contact-form/class-contact-form.php#L1072
